src/libraries/System.Diagnostics.DiagnosticSource/tests/ActivitySourceTests.cs

```diff
@@ -382,6 +382,29 @@ public void TestDefaultParentContext()
             }).Dispose();
         }
 
+        [Fact]
+        public void TestActivityContextParsing()
+        {
+            const string w3cId = "00-99d43cb30a4cdb4fbeee3a19c29201b0-e82825765f051b47-01";
```

Comment:
can you add a test passing a non-valid w3cId as input?

Reply:
I added a test. Thanks.